lerobot/common/robot_devices/robots/trossen_ai_mobile.py

```diff
@@ -0,0 +1,387 @@
+            images[name] = torch.from_numpy(images[name])
+            self.logs[f"read_camera_{name}_dt_s"] = self.cameras[name].logs["delta_timestamp_s"]
+            self.logs[f"async_read_camera_{name}_dt_s"] = time.perf_counter() - before_camread_t
+
+        # Populate output dictionaries and format to pytorch
+        obs_dict = {}
+        obs_dict["observation.state"] = state
+        for name in self.cameras:
+            obs_dict[f"observation.images.{name}"] = images[name]
+
+        obs_dict = {}
+        obs_dict["observation.state"] = state
+
+        return obs_dict
+
+    def send_action(self, action):
+        if not self.is_connected:
+            raise RobotDeviceNotConnectedError(
+                "TrossenAIMobileRobot is not connected. You need to run `robot.connect()`."
+            )
+
+        from_idx = 0
+        to_idx = 0
+        action_sent = []
+        for name in self.follower_arms:
+            # Get goal position of each follower arm by splitting the action vector
+            to_idx += len(self.follower_arms[name].motor_names)
+            goal_pos = action[from_idx:to_idx]
+            from_idx = to_idx
+
+            # Cap goal position when too far away from present position.
+            # Slower fps expected due to reading from the follower.
+            if self.config.max_relative_target is not None:
+                present_pos = self.follower_arms[name].read("Present_Position")
+                present_pos = torch.from_numpy(present_pos)
+                goal_pos = ensure_safe_goal_position(goal_pos, present_pos, self.config.max_relative_target)
+
+            # Save tensor to concat and return
+            action_sent.append(goal_pos)
+
+            # Send goal position to each follower
+            goal_pos = goal_pos.numpy().astype(np.float32)
+            self.follower_arms[name].write("Goal_Position", goal_pos)
+
+        linear_vel, angular_vel = action.tolist()[-2:]
+        before_write_t = time.perf_counter()
+        self.base.set_cmd_vel(linear_vel, angular_vel)
+        self.logs["write_base_dt_s"] = time.perf_counter() - before_write_t
+
+        action_sent.append(action[-2:])
+
+        return torch.cat(action_sent)
+
+    def disconnect(self):
+        if not self.is_connected:
+            raise RobotDeviceNotConnectedError(
+                "TrossenAIMobileRobot is not connected. You need to run `robot.connect()` before disconnecting."
+            )
+        
+        self.base.enable_motor_torque(False)
+
+        for name in self.follower_arms:
+            self.follower_arms[name].disconnect()
+
+        for name in self.leader_arms:
+            self.leader_arms[name].disconnect()
+        
+        time.sleep(2)
+
+        for name in self.cameras:
+            self.cameras[name].disconnect()
+        
+        self.is_connected = False
+
+    def __del__(self):
+        if getattr(self, "is_connected", False):
+            self.disconnect()
```

Comment:
newline at end of file

```suggestion
            self.disconnect()

```